tests/scripts/all.sh

```diff
@@ -4626,8 +4626,7 @@ component_test_m32_o0 () {
     # build) and not the i386-specific inline assembly.
     msg "build: i386, make, gcc -O0 (ASan build)" # ~ 30s
     scripts/config.py full
-    scripts/config.py unset MBEDTLS_AESNI_C # AESNI depends on cpu modifiers
-    make CC=gcc CFLAGS="$ASAN_CFLAGS -m32 -O0" LDFLAGS="-m32 $ASAN_CFLAGS"
+    make CC=gcc CFLAGS="$ASAN_CFLAGS -m32 -O0 -maes -msse2 -mpclmul" LDFLAGS="-m32 $ASAN_CFLAGS"
```

Comment:
Why is this change being made?

The comment is clear: this component is intended to test the portable C code in a 32-bit build

Reply:
I was trying to revert the workarrounds added in #7384, that's why I removed the previous line 4629, and the additional CFLAGS were just for a successful build. 
I have just realize that all AESNI m32 test are in a separate component, so only reverting the work arround in `component_build_mingw` should be sufficient?

BTW, I don't quite understand the comments in line 4625 and line 4642. Why different optimization levels make a difference on using or not using i386-specific inline assembly?